log/with_logger.go

```diff
@@ -57,7 +52,7 @@ type withLogger struct {
 }
 
 func newWithLogger(logger Logger, keyvals ...interface{}) *withLogger {
-	return &withLogger{logger: logger, keyvals: keyvals}
+	return &withLogger{logger: Skip(logger, 1), keyvals: keyvals}
```

Comment:
Just to confirm, this code path would only matter if the logger did not implement `WithLogger` but did implement `WithSkipCallers`?

Reply:
Yes, you right.
`Skip` function will ignore logger in case it doesn't implement `WithSkipCallers` interface

Reply:
hm I wonder how common this actually is, but in principle this change makes sense I think